app/src/main/java/com/neura/sampleapplication/NeuraManager.java

```diff
@@ -46,10 +63,160 @@ public NeuraApiClient getClient() {
      * the settings). Fyi this only means that the sensors are disabled, not when there's no wifi available fe.
      */
     public void initNeuraConnection(Context context) {
+        if (!isMinVersion()) {
+            return;
+        }
+
         Builder builder = new Builder(context);
         mNeuraApiClient = builder.build();
         mNeuraApiClient.setAppUid(context.getResources().getString(R.string.app_uid));
         mNeuraApiClient.setAppSecret(context.getResources().getString(R.string.app_secret));
         mNeuraApiClient.connect();
     }
-}
+
+    //create a call back to handle authentication stages.
+    private static AnonymousAuthenticationStateListener silentStateListener = new AnonymousAuthenticationStateListener() {
+        @Override
+        public void onStateChanged(AuthenticationState state) {
+
+            switch (state) {
+                case AccessTokenRequested:
+                    break;
+                case AuthenticatedAnonymously:
+                    // successful authentication
+                    NeuraManager.getInstance().getClient().unregisterAuthStateListener();
+
+//                    NeuraManager.getInstance().getClient().getUserDetails(new UserDetailsCallbacks() {
```

Comment:
Let's make sure that getting the user's Neura ID and the user's Access Token is done in a different method